### ISSUE TITLE
oci_tarball for multiarch image fails with `blobs/json: no such file or directory`

### ISSUE THREAD

**Issue by exp0nge** [bazel-contrib/rules_oci#651]
rules_oci version: 1.7.6
Runtimes:
* M1 with containerd enabled
* Linux/amd64: runc

Snippet of the tarball:

```bazel
oci_image(
    name = "image-linux_x86_64",
    base = "@bookworm_base_linux_amd64",
    cmd = ["./inux_x86_64"],
    tars = [":tar-linux_x86_64"],
    visibility = ["//visibility:public"],
    workdir = "/",
)

oci_image(
    name = "image-linux_arm64",
    base = "@bookworm_base_linux_arm64_v8",
    cmd = ["./inux_arm64"],
    tars = [":tar-linux_arm64"],
    visibility = ["//visibility:public"],
    workdir = "/",
)

oci_image_index(
    name = "image-multiarch",
    images = [
        ":image-linux_arm64",
        ":image-linux_x86_64",
    ],
)

oci_tarball(
    name = "tarball_multiarch",
    format = "oci",
    image = ":image-multiarch",
    repo_tags = ["multiarch:latest"],
)
```

This ends up with

```
❯ bazel run src/app:tarball_multiarch
open /var/lib/docker/tmp/docker-import-1480447137/blobs/json: no such file or directory
```

Tried directly:

```
❯ docker load < bazel-bin/src/app/tarball_multiarch/tarball.tar
open /var/lib/docker/tmp/docker-import-2417380582/blobs/json: no such file or directory
```

**Comment by thesayyn** [comment]
It's possible that this is already fixed, could try and see if this is present in https://github.com/bazel-contrib/rules_oci/releases/tag/v1.8.0?

**Comment by ekhabarov** [comment]
Encountered the same issue with the tarball target where `format="oci"` and `image = ":image-index"`. Tried v1.8.0 but it has another issue #653 at least on Mac M2.

**Comment by ekhabarov** [comment]
> It's possible that this is already fixed, could try and see if this is present in https://github.com/bazel-contrib/rules_oci/releases/tag/v1.8.0?

Tried v1.8.0 without #602, still the same error.

**Comment by exp0nge** [comment]
> ```python
>     format = "oci",
> ```

Same error as v1.8.0

**Comment by thesayyn** [comment]
> Tried v1.8.0 without #602, still the same error.

Are these two related, because #602 is about oci_pull whereas this issue is about oci_tarball.

**Comment by ekhabarov** [comment]
> > Tried v1.8.0 without #602, still the same error.
> 
> Are these two related? because #602 is about oci_pull whereas this issue is about oci_tarball.

I don't think so. I mentioned it because version upgrade 1.7.6 => 1.8.0 brought different error.

**Comment by chulkilee** [comment]
It still fails on the main branch at https://github.com/bazel-contrib/rules_oci/blob/a32f3bfe82464cda39959a6ae24b8ec547c67abc/oci/private/load.sh.tpl#L31

Seems like `docker load` requires docker-specific metadata which is not present in oci formats.

If docker format itself supports multi-arch images, then probably we have to implement that in rules_oci - probably at oci/`private/tarball.sh.tpl`

**Comment by chulkilee** [comment]
docker engine 27.2.0 seems to support loading oci image but I still get the same error.

https://github.com/moby/moby/pull/44833

You can test it by adding followings to `examples/multi_architecture_image/BUILD.bazel` and run `bazel run //examples/multi_architecture_image:load`

```bazel
oci_load(
    name = "load",
    image = ":index",
    repo_tags = ["my-repository:latest"],
    format = "oci",
)
```

**Comment by thesayyn** [comment]
Docker by default does not support loading oci tarballs into the daemon, that's probably what you are seeing here.  Please make sure that you enabled containerd image store experiment if you are using docker.

https://docs.docker.com/desktop/containerd/

**Comment by thesayyn** [comment]
This issue is now a documentation issue. Need to mention enabling containerd image store.

**Comment by exp0nge** [comment]
> This issue is now a documentation issue. Need to mention enabling containerd image store.

I reported this issue and used containerd in darwin/arm64 and debian/amd64. Has it been fixed when using containerd?

**Comment by thesayyn** [comment]
> Has it been fixed when using containerd

It should work with containerd.

**Comment by alemoreno991** [comment]
From the [documentation](https://docs.docker.com/engine/storage/containerd/#:~:text=Enable%20containerd%20image%20store%20on%20Docker%20Engine,take%20effect.%20$%20sudo%20systemctl%20restart%20docker.) we can see what @thesayyn mentions: 

> While the `overlay2` driver still remains the default driver for Docker Engine, you can opt in to using containerd snapshotters as an experimental feature


I use NixOS and since I faced this problem I'd like to share how to solve it for that OS.
From some exploration one arrives to the NixOS Packages Options documetation where the [virtualisation.docker.daemon.settings](https://search.nixos.org/options?channel=24.11&show=virtualisation.docker.daemon.settings&from=0&size=50&sort=relevance&type=packages&query=virtualisation.docker) allows us to write the `daemon.json` file. 

In conclusion, you need to add the following to your `configuration.nix` file:

```
  virtualisation = {
    containerd.enable = true;
    docker = {
      enable = true;
      daemon.settings = {
        features = {
          containerd-snapshotter = true;
        };
      };
    };
  };

  users.users.your-username.extraGroups = ["docker"];
```

**Comment by alexeagle** [comment]
@alemoreno991 do you have a few minutes to send a PR to the docs here, in whatever spot you think would make this discoverable?

**Comment by alemoreno991** [comment]
Addressed by #806
